docs/externalnode.md

```diff
@@ -0,0 +1,309 @@
+
+#### IP packet
+
+One simple OpenFlow pipeline is set on an external Node to process the IP packet.
+Antrea NetworkPolicy is the major functions in this new pipeline, and the OpenFlow
+tables used are similar to the container case. **No L3 routing is provided on an
+external Node**, and a simple L2 forwarding policy is implemented. OVS conntrack is
+used to enforce the NetworkPolicy rule on a connection but not on every single
+packet.
+
+**Egress/Ingress Tables**
+
+Table `XgressSecurityClassifierTable` is installed in both stageEgressSecurity and
+stageIngressSecurity, which is used to install the OpenFlow entries to ensure some
+specific IP traffic to bypass Antrea NetworkPolicy rules, e.g.,the SSH connection
+from a specific host to the external Node. These rules can be set with the item
+`policyBypassRules` in the agent configuration. Following is a configuration example,
+
+```yaml
+policyBypassRules: "in:tcp:1.1.1.1:22"
+```
+
+This is an example of the OpenFlow entry for the above configuration,
+
+```yaml
+table=IngressSecurityClassifier, priority=200,ct_state=+new+trk,tcp,nw_src=1.1.1.1,tp_dst=22 actions=resubmit(,IngressMetric)
+```
+
+Other OpenFlow tables in stageEgressSecurity and stageIngressSecurity are the same
+as those installed on Kubernetes worker Node.
+
+More details please refer to the general [introduction](design/ovs-pipeline.md) of
+Antrea OVS pipeline.
+
+**L2 Forwarding Tables**
+
+`L2ForwardingCalcTable` is used to calculate the expected output port of an IP packet.
+As the pair ports with the internal port and uplink always exist on the OVS bridge,
+and both two interfaces are configured with the same MAC address, the match condition
+of the OpenFlow entry in `L2ForwardingCalcTable` is the input port number but not
+the MAC address of the packet. The actions include, 1) set flag `OFPortFoundRegMark`,
+and set the peer port as the `TargetOFPortField`, and then 2) enforce the packet to
+go to stageIngressSecurity.
+
+Following is an example of the OpenFlow entries in `L2ForwardingCalcTable`
+
+```yaml
+table=L2ForwardingCalc, priority=200,ip,in_port=ens224 actions=load:0x1->NXM_NX_REG0[8],load:0x7->NXM_NX_REG1[],resubmit(,IngressSecurityClassifier)
+table=L2ForwardingCalc, priority=200,ip,in_port="phy-ens224" actions=load:0x1->NXM_NX_REG0[8],load:0x8->NXM_NX_REG1[],resubmit(,IngressSecurityClassifier)
+```
+
+## RBAC
+
+An external Node is thought as an untrusted point on the network. To follow the
+security principle that least permissions are given, the RBAC for antrea-agent
+running with `ExternalNode` feature is modified:
+- Only `get`, `list` and `watch` permissions are given on resource `ExternalNode`
+- Only `update` permission is given on resource `antreaagentinfos`, and `create`
+  permission is moved to antrea-controller
+- other permissions please refer to [vm-agent.yaml](../build/yamls/externalnode/vm-agent.yaml)
+
+antrea-agent still uses `antreaagentinfo` object to report its status on `ExternalNode`.
```

Comment:
antrea-agent still uses `antreaagentinfo`
Don't follow what still means here? Probably rephrase sentence

Reply:
Rephrase it.